### PR TITLE
CIRC-4638 - Add filters/show Endpoint

### PR DESCRIPTION
**Pull request by pamaddox**
Add a new endpoint - "filters/show", that functions like the existing "checks/show"
endpoint. It returns an XML list of all filtersets.

### CODE REVIEW

File: src/noit_filters_lmdb.c [hunk 1]
```diff
@@ -1231,6 +1231,69 @@ noit_filters_lmdb_process_repl(xmlDocPtr doc) {
...
+  goto cleanup;
+
+ error:
+  mtev_http_response_standard(ctx, error_code, "ERROR", "text/html");
+  mtev_http_response_end(ctx);
+  goto cleanup;
+
+ cleanup:
+  if(doc) {
+    xmlFreeDoc(doc);
```
Comment (pjulien) [inline src/noit_filters_lmdb.c:1291]: Is this reference counted or does ``mtev_http_response_xml`` render the entire document to a string?

Comment (pamaddox) [reply src/noit_filters_lmdb.c:1291]: `mtev_http_response_xml` returns a string to the caller

File: src/noit_filters_lmdb.c [hunk 2]
```diff
@@ -1231,6 +1231,69 @@ noit_filters_lmdb_process_repl(xmlDocPtr doc) {
...
+
+  mtev_http_response_ok(ctx, "text/xml");
+  mtev_http_response_xml(ctx, doc);
+  mtev_http_response_end(ctx);
+  goto cleanup;
+
+ error:
+  mtev_http_response_standard(ctx, error_code, "ERROR", "text/html");
+  mtev_http_response_end(ctx);
+  goto cleanup;
```
Comment (vasuraman11) [inline src/noit_filters_lmdb.c:1287]: nit - unneeded, cleanup is the next source line

Comment (pamaddox) [reply src/noit_filters_lmdb.c:1287]: Here for consistency. The existing endpoints also do this.

**Pull request merged**